app/models/user.rb

```diff
@@ -1326,45 +1312,15 @@ def can_manage_delegate_probation?
   end
 
   def senior_delegates
-    active_roles
-      .select { |role| UserRole.is_group_type?(role, UserGroup.group_types[:delegate_regions]) }
-      .map { |role| UserRole.group(role).senior_delegate }
+    delegate_roles.map { |role| role.group.senior_delegate }
   end
 
   def regional_delegates
-    active_roles
-      .select { |role| UserRole.is_group_type?(role, UserGroup.group_types[:delegate_regions]) }
-      .map { |role| UserRole.group(role).lead_user }
-  end
-
-  def active_roles
-    roles.select { |role| UserRole.is_active?(role) }
+    delegate_roles.map { |role| UserRole.group(role).lead_user }
```

Comment:
Here you're inconsistent about using `UserRole.group` helper vs. actual `group` association.
I would prefer using the `UserRole.group` helper above, because that will give us easier access to find all places we can optimize at the end of this project.

Reply:
Done